04-onion-routing.md

```diff
@@ -601,6 +606,20 @@ If the `cltv-expiry` is too low, the final node MUST fail the HTLC:
 
 1. type: 17 (`final_expiry_too_soon`)
 
+If the `outgoing_cltv_value` does not match the `ctlv-expiry` of the
+HTLC at the final hop:
+
+1. type: 18 (`final_incorrect_cltv_expiry`)
+2. data:
+   * [4:cltv-expiry]
+
+If the `amt_to_forward` does not match the `incoming_htlc_amt` of
```

Comment:
Oh, I see your suggestion: that we put 'fee' in the onion instead of 'amt_to_forward'?

Consider A->B->C->D.

The advantage of `amt_to_forward` is that if node B takes more fee than expected, node C detects it immediately, and returns an error, blaming B.  If C only knows fee, it subtracts it and then D returns an error (insufficient payment), and A has no idea which node took too much...